src/Tasks/Microsoft.NET.Build.Tasks/GenerateShims.cs

```diff
@@ -0,0 +1,173 @@
+                var appBinaryFilePath =
+                    $".store/{PackageId.ToLowerInvariant()}/{PackageVersion}/{PackageId.ToLowerInvariant()}/{PackageVersion}/tools/{targetFramework.GetShortFolderName()}/any/{ToolEntryPoint}";
+
+                Directory.CreateDirectory(packagedShimOutputDirectoryAndRid);
+                if (File.Exists(appHostDestinationFilePath))
+                {
+                    File.Delete(appHostDestinationFilePath);
+                }
+
+                AppHost.Create(
+                    resolvedApphostAssetPath,
+                    appHostDestinationFilePath,
+                    appBinaryFilePath
+                );
+
+                var item = new TaskItem(appHostDestinationFilePath);
+                item.SetMetadata("ShimRuntimeIdentifier", runtimeIdentifier);
+                embeddedApphostPaths.Add(item);
+            }
+
+            EmbeddedApphostPaths = embeddedApphostPaths.ToArray();
+        }
+
+        private string GetApphostAsset(NuGetFramework targetFramework, LockFile lockFile, string runtimeIdentifier)
+        {
+            var apphostName = DotNetAppHostExecutableNameWithoutExtension;
+
+            if (runtimeIdentifier.StartsWith("win"))
+            {
+                apphostName += ".exe";
+            }
+
+            LockFileTarget runtimeTarget = lockFile.GetTargetAndThrowIfNotFound(targetFramework, runtimeIdentifier);
+
+            return FindApphostInRuntimeTarget(apphostName, runtimeTarget);
+        }
+
+        private string FindApphostInRuntimeTarget(string apphostName, LockFileTarget runtimeTarget)
+        {
+            foreach (LockFileTargetLibrary library in runtimeTarget.Libraries)
+            {
+                if (!library.IsPackage())
+                {
+                    continue;
+                }
+
+                foreach (LockFileItem asset in library.NativeLibraries)
+                {
+                    if (asset.IsPlaceholderFile())
+                    {
+                        continue;
+                    }
+
+                    var resolvedPackageAssetPath = ResolvePackageAssetPath(library, asset.Path);
+                    if (Path.GetFileName(resolvedPackageAssetPath) == apphostName)
+                    {
+                        return resolvedPackageAssetPath;
+                    }
+                }
+            }
+
+            throw new BuildErrorException(Strings.CannotFindApphostForRid, runtimeTarget.RuntimeIdentifier);
+        }
+
+        private string ResolvePackageAssetPath(LockFileTargetLibrary package, string relativePath)
+        {
+            var packagePath = _packageResolver.GetPackageDirectory(package.Name, package.Version);
+            return Path.Combine(packagePath, NormalizeRelativePath(relativePath));
+        }
+
+        private static string NormalizeRelativePath(string relativePath)
+        {
+            return relativePath.Replace('/', Path.DirectorySeparatorChar);
+        }
```

Comment:
This looks copied from ResolvePackageAssets before [this](https://github.com/dotnet/sdk/commit/b9d8bcf65d67c7cf9049fe104eea69bbf0aeaad2#diff-42d31913aec3dcadaa9bf9b5eafba94cR44) and [this](https://github.com/dotnet/sdk/commit/d6b5c8095f381744c90db95b4dfe9d6369d2cee3#diff-42d31913aec3dcadaa9bf9b5eafba94cR48)

You should remove this and call ResolvePackageAssetPath on the resolver.

Reply:
Fixed, removed all these and just call nugetpackageresolver